### ISSUE TITLE
How to add tooltip for every item in hamburger

### ISSUE THREAD

**Issue by a516117636** [MahApps/MahApps.Metro#2928]
I want to show tooltip when the pane is closed because there is not any prompt message but a icon,I did not find the way,can you help me? Thanks!

**Comment by punker76** [comment]
@a516117636 One way is to create your own menu item with a property which can you use at the DataTemplate to show a ToolTipp. I've update my samples, just take a look here https://github.com/punker76/code-samples/tree/master/MahAppsMetroHamburgerMenu

```csharp
public class CustomHamburgerMenuIconItem : HamburgerMenuIconItem
{
    public static readonly DependencyProperty ToolTipProperty
        = DependencyProperty.Register("ToolTip",
            typeof(object),
            typeof(CustomHamburgerMenuIconItem),
            new PropertyMetadata(null));

    public object ToolTip
    {
        get { return (object)GetValue(ToolTipProperty); }
        set { SetValue(ToolTipProperty, value); }
    }
}
```

```xaml
<DataTemplate x:Key="MenuItemTemplate" DataType="{x:Type local:CustomHamburgerMenuIconItem}">
    <Grid x:Name="RootGrid"
            Height="48"
            Background="Transparent">
        <Grid.ColumnDefinitions>
            <ColumnDefinition Width="48" />
            <ColumnDefinition />
        </Grid.ColumnDefinitions>
        <ContentControl Grid.Column="0"
                        HorizontalAlignment="Center"
                        VerticalAlignment="Center"
                        Content="{Binding Icon}"
                        Focusable="False"
                        Foreground="White" />
        <TextBlock Grid.Column="1"
                    VerticalAlignment="Center"
                    FontSize="16"
                    Foreground="White"
                    Text="{Binding Label}" />
    </Grid>
    <DataTemplate.Triggers>
        <DataTrigger Binding="{Binding RelativeSource={RelativeSource FindAncestor, AncestorType={x:Type controls:HamburgerMenu}}, Path=IsPaneOpen}" Value="False">
            <Setter TargetName="RootGrid" Property="ToolTip" Value="{Binding ToolTip, Mode=OneWay}" />
        </DataTrigger>
    </DataTemplate.Triggers>
</DataTemplate>
```

usage

```xaml
<local:CustomHamburgerMenuIconItem Icon="{iconPacks:PackIconMaterial Kind=Home}"
                                    Label="Home"
                                    ToolTip="Show the Home view">
    <local:CustomHamburgerMenuIconItem.Tag>
        <views:HomeView />
    </local:CustomHamburgerMenuIconItem.Tag>
</local:CustomHamburgerMenuIconItem>
```

**Comment by avechuche** [comment]
@punker76 Will this option be available in version 1.6.0? Thank you!

**Comment by a516117636** [comment]
thank you very much!
